### PR TITLE
Test/dec 20 vivek

### PR DESCRIPTION
**Pull request by vivek1214-git**
completed- form update
todo- form not showing ,view page is  not showing and cant redirect to the new form

### CODE REVIEW

**Comment by vivek1214-git** [comment]
completed country form and tag form delete functionality
todo  add new data insert msg functionality remaining.

File: app/Http/Controllers/CountriesController.php [hunk 1]
```diff
@@ -0,0 +1,65 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Models\User;
+use App\Models\Country;
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\Session;
+
+class CountriesController extends Controller
```
Comment (ithqindia) [inline app/Http/Controllers/CountriesController.php:10]: CountryController

File: app/Http/Controllers/CountriesController.php [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+{
+    public function index()
+    {
+        $countryInformation = Country::get();
```
Comment (ithqindia) [inline app/Http/Controllers/CountriesController.php:14]: $countryInformation instead use $countries

File: app/Http/Controllers/CountriesController.php [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+            'is_enabled' => (bool) $request->get('is_enabled'),
+        ]);
+        return view('success-country');
+    }
+
+    public function show(Request $request)
+    {
+        $countryInformation = Country::where('id', $request->id)->get()->first();
+        $user = [];
+        //dd(CountryInformation);
```
Comment (ithqindia) [inline app/Http/Controllers/CountriesController.php:36]: Remove these lines

File: app/Http/Controllers/SelfieController.php [hunk 4]
```diff
@@ -14,7 +15,7 @@ public function store(Request $request)
             'selfie' => 'required',
         ]);
 
-        Selfie::create([
+        Selfies::create([
```
Comment (ithqindia) [inline app/Http/Controllers/SelfieController.php:18]: Check naming conventions https://ithqworkspace.slack.com/archives/C02H79MJNQG/p1640177593015700
Model names should be singular 
Update at all places

File: app/Http/Controllers/TagController.php [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+            'name' => $request->get('name'),
+
+        ]);
+        return view('success-tag');
+    }
+    public function show(Request $request)
+    {
+        $tag = Tag::where('id', $request->id)->get()->first();
+        $user =[];
+        //dd(CountryInformation);
```
Comment (ithqindia) [inline app/Http/Controllers/TagController.php:31]: remove these lines

File: app/Models/Country.php [hunk 6]
```diff
@@ -11,7 +11,7 @@ class Country extends Model
     use HasFactory;
 
     protected $guarded = [];
-
+    protected $table = "Countries";
```
Comment (ithqindia) [inline app/Models/Country.php:14]: Remove this line. Not required to specify table name

File: app/Models/Tag.php [hunk 7]
```diff
@@ -11,7 +11,7 @@ class Tag extends Model
     use HasFactory;
 
     protected $guarded = [];
-
+    protected $table = "tags";
```
Comment (ithqindia) [inline app/Models/Tag.php:14]: Remove this line. Not required to specify table name

File: database/factories/CounselorSessionFactory.php [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+     *
+     * @return array
+     */
+    public function definition()
+    {
+        return [
+            'session_date' => $this->faker->date('d_m_Y'),
+            'session_time' => $this->faker->time('H_i_s'),
+            'meeting_name' => $this->faker->randomElement(['online', 'offline']),
+            //'address2' => $this->faker->address(),
```
Comment (ithqindia) [inline database/factories/CounselorSessionFactory.php:21]: remove this line

File: database/factories/ExtraScoreFactory.php [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+     *
+     * @return array
+     */
+    public function definition()
+    {
+        return [
+            'exam_date' => $this->faker->date('d_m_Y'),
+            'score' => $this->faker->randomFloat(),
+           // 'meeting_name' => $this->faker->randomElement(['online', 'offline']),
+            //'address2' => $this->faker->address(),
```
Comment (ithqindia) [inline database/factories/ExtraScoreFactory.php:21]: remove these lines

File: database/factories/SelfieFactory.php [hunk 10]
```diff
@@ -0,0 +1,25 @@
...
+     *
+     * @return array
+     */
+    public function definition()
+    {
+        return [
+            'selfie' => 'https://source.unsplash.com/random',
+            //'score' => $this->faker->randomFloat(),
+           // 'meeting_name' => $this->faker->randomElement(['online', 'offline']),
+            //'address2' => $this->faker->address(),
```
Comment (ithqindia) [inline database/factories/SelfieFactory.php:21]: remove these lines

File: routes/web.php [hunk 11]
```diff
@@ -170,3 +180,41 @@
...
+
+Route::get('/country', function () {
+     $countries = Country::get()->sortBy('name');
+    return view('country', compact('countries'));
+ });
+
+ Route::get('/tag', function () {
+    $tags = Tag::get()->sortBy('name');
+    return view('tag', compact('tags'));
+});
```
Comment (ithqindia) [inline routes/web.php:220]: These are not required. 
Following routes do the same thing
Route::get('/tag', [TagController::class, 'index']);
Route::get('/country', [CountriesController::class, 'index']);

**Review by ithqindia — CHANGES_REQUESTED** [review]

File: app/Http/Controllers/CountryController.php [hunk 12]
```diff
@@ -7,12 +7,12 @@
...
 use Illuminate\Support\Facades\Session;
 
-class CountriesController extends Controller
+class CountryController extends Controller
 {
     public function index()
     {
-        $countryInformation = Country::get();
-        return view('country', compact('countryInformation'));
+        $country = Country::get();
```
Comment (ithqindia) [inline app/Http/Controllers/CountryController.php:14]: We are getting all data from Country so the variable would be $countries

File: app/Http/Controllers/CountryController.php [hunk 13]
```diff
@@ -26,40 +26,40 @@ public function store(Request $request)
...
-        //dd(CountryInformation);
-        return view('country', compact('countryInformation', 'user'));
+        $country = Country::where('id', $request->id)->get()->first();
+        return view('/country', compact('country'));
     }
 
     public function update(Request $request)
     {
-        $countryInformation = Country::find($request->id);
+        $countries = Country::find($request->id);
```
Comment (ithqindia) [inline app/Http/Controllers/CountryController.php:42]: Here we are getting single country hence variable $country

File: app/Http/Controllers/CountryController.php [hunk 14]
```diff
@@ -26,40 +26,40 @@ public function store(Request $request)
...
         Country::find($request->id)->delete();
         Session::flash('message', 'Data deleted successfully !');
-        return view('country-info');
+        return redirect('/show-country');
     }
     public function view(Request $request)
     {
-        $countryInformation = Country::find($request->id);
-        return view('country-info', compact('countryInformation'));
+        $countries = Country::find($request->id);
```
Comment (ithqindia) [inline app/Http/Controllers/CountryController.php:62]: Here we are getting single country hence variable $country

File: routes/web.php [hunk 15]
```diff
@@ -101,6 +102,31 @@
...
+
+Route::get('/show-country', function () {
+ $country = Country::get()->sortBy('name');
+    return view('show-country', compact('country'));
+});
+
+Route::get('/show-tag', function () {
+    $tags = Tag::get()->sortBy('name');
+    return view('show-tag', compact('tags'));
+ });
```
Comment (ithqindia) [inline routes/web.php:123]: Need to update as per https://ithqworkspace.slack.com/archives/C02H79MJNQG/p1640252019001700

File: routes/web.php [hunk 16]
```diff
@@ -101,6 +102,31 @@
...
+
+
+
+
+
+
```
Comment (ithqindia) [inline routes/web.php:129]: Do not have more than one empty line

**Review by ithqindia — CHANGES_REQUESTED** [review]

**Comment by ithqindia** [comment]
Amol is managing this branch. Vivek's branch is https://github.com/ithqindia/metroclub/pull/17
both these branches should be merged into develop

**Pull request closed without merging**